libs/sdk-core/src/breez_services.rs

```diff
@@ -2606,12 +2613,34 @@ impl PaymentReceiver {
 async fn get_lsp(
     persister: Arc<SqliteStorage>,
     lsp_api: Arc<dyn LspAPI>,
+    skip_cache: bool,
 ) -> Result<LspInformation> {
-    let lsp_id = persister.get_lsp_id()?.ok_or(anyhow!("No LSP ID found"))?;
-
-    get_lsp_by_id(persister, lsp_api, lsp_id.as_str())
-        .await?
-        .ok_or_else(|| anyhow!("No LSP found for id {lsp_id}"))
+    let persisted_lsp_info =
+        persister.get_lsp_information()?.and_then(|lsp_info| {
+            match !skip_cache && lsp_info.opening_fee_params_list.is_valid() {
+                true => Some(lsp_info),
+                false => {
+                    debug!("Ignoring LSP information: {:?} {:?}", skip_cache, lsp_info);
+                    None
+                }
+            }
+        });
+    match persisted_lsp_info {
+        Some(lsp_info) => {
+            debug!("Using persisted LSP information");
+            Ok(lsp_info)
+        }
+        None => {
```

Comment:
```suggestion
    let persisted_lsp_info = persister.get_lsp_information()?;
    let cached_lsp_info = match skip_cache {
        true => None,
        false => persisted_lsp_info.filter(|lsp_info| lsp_info.opening_fee_params_list.is_valid()),
    };

    match cached_lsp_info {
        Some(lsp_info) => {
            debug!("Using cached LSP info");
            Ok(lsp_info)
        }
        None => {
            debug!("Ignoring cached LSP info: skip_cache={skip_cache} lsp_info={lsp_info:?}");
```

Reply:
Added a suggestion on how the first part of the method could be rewritten for clarity.